### ISSUE TITLE
ConsumerWidget not working as expected on Android after changing compileSdkVersion to 31

### ISSUE THREAD

**Issue by playfairdisplay** [rrousselGit/river_pod#1318]
**Describe the bug**
State changes don't update the UI on Android after changing compileSdkVersion and targetSdkVersion to 31. The iOS app works completely fine.
flutter_riverpod version is 0.14.0+3
**To Reproduce**
Use flutter_riverpod version 0.14.0+3 and set compileSdkVersion to 31 and check on Android, the ConsumerWidget doesn't update the UI when state changes. However, on hot reload, the UI is updated (basically hot reload changes the UI to latest state).

**Expected behavior**
I expected the app to update the UI per state change on Android as it's working fine on iOS.

**Comment by rrousselGit** [comment]
Could you include a way to reproduce the error?

It's quite likely that the problem is elsewhere and likely a bug in your code.

My bet is that you upgraded your dependency on state_notifier implicitly, which released a breaking change.

Remember that you're not supposed to mutate the state of a StateNotifier, it should be immutable.

**Comment by playfairdisplay** [comment]
Thanks for your comment. After you pointed out state_notifier might be playing a role here, I found a similar issue here on the state_notifier's repo: https://github.com/rrousselGit/state_notifier/issues/58

Due to the newly added 'updateShouldNotify' method, the state didn't update as it was always returning false.  For now, I've reverted back to the 0.7.0 version that doesn't introduce this method, and it's working fine again. I'll soon update my code to work with the method.
Thanks for the help!
